backend/routes/userRoute.js

```diff
@@ -17,11 +24,46 @@ router.get('/api/user', (req, res) => {
 });
 
 router.post('/api/verify-credentials', isAuthenticated, (req, res) => {
-  const { username } = req.body;
-  console.log(`processing verification for ${username}`);
+  const { username, mongoId } = req.body;
+  console.log(`processing verification for ${username}`, mongoId);
   // process verification with FCC here ...
-
-  res.sendStatus(201);
+  axios.all([getFrontEndCert(username), getBackEndCert(username), getDataVisCert(username)])
+  .then(axios.spread((frontCert, backCert, dataCert) => {
+    
+    if (frontCert.request._redirectCount === 0) {
+      User.findByIdAndUpdate(mongoId, { 
+        '$push' : { fccCerts: "Front End Certification Earned!!" },
+        '$set' : { username: username }
+      },
+      function(err, user) { if (err) throw err });
+    } else {
+      // updating username if they entered a new one, even if no cert owned
+      User.findByIdAndUpdate(mongoId, { '$set' : { username: username }}, function(err, user) {
+        if (err) throw err;
+      });
+    }
+    
+    if (backCert.request._redirectCount === 0) {
+      User.findByIdAndUpdate(mongoId, { 
+        '$push' : { fccCerts: "Back End Certification Earned!!" }
+      }, function(err, user) { if (err) throw err });
+    } 
+    
+    if (dataCert.request._redirectCount === 0) {
+      User.findByIdAndUpdate(mongoId, { 
+        '$push' : { fccCerts: "Data Visualization Certification Earned!!" }
+      }, function(err, user) { if (err) throw err });
+    } 
+  })).then(
+    // even though user is already updated by the time of this
+    // .then(), what's returned is the document before the update
+    // so info is not current, we need this data to be current
+    // seems like it should work, not sure why it doesn't 
```

Comment:
I'm not sure exactly why you wouldn't be getting the updated user at the end of this... but why not just create one array/object based on the resolution of these three calls to FCC and then find & save the user once? As in update something based on the redirect counts to produce something looking like this:

```
var certs = {
   front: true,
   data: false,
   back: true
}
```

and then save the user and return, e.g.

```
user.fccCerts = certs;
user.save();
res.json({ user });
```